lib/diagnostics_channel.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+const {
+  ArrayPrototypeIncludes,
+  ArrayPrototypePush,
+  FunctionPrototypeBind,
+  ObjectCreate,
+  ObjectSetPrototypeOf,
+} = primordials;
+
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+  }
+} = require('internal/errors');
+
+// TODO(qard): should there be a C++ publisher interface?
```

Comment:
I think this is needed if you plan to publish from C++.

Reply:
Yep, just not really sure yet if there's anything we actually _need_ from the C++ side. I'll try to discuss more in the diagnostics working group meeting to see if anyone else has any thoughts on that.

Reply:
No cases for a C++ API have come up yet so I'm going to mark this resolved. If demand for it comes up in the future, I can consider it then--no point adding it if no one needs it. 🤷